reference/implementing-a-concept-exercise.md

```diff
@@ -0,0 +1,89 @@
+
+Because CIS 194 may not cover all subject that contributors wish to cover in this material, extending the concepts beyond the CIS 194 course should ideally be accompanied by some level of reasoning about the curriculum.
+
+The directory structure of a single concept exercise looks like this:
+
+```
+languages
+└── haskell
+    ├── concepts
+    |   └── <CONCEPT_SLUG>
+    |       ├── about.md
+    |       └── links.json
+    └── exercises
+        └── concept
+            └── <SLUG>
+                ├── package.yaml
+                ├── stack.yaml
+                ├── src
+                │   └── <NAME>.hs
+                ├── test
+                │   └── Tests.hs
+                ├── exemplars
+                │   ├── success-... (default: standard)
+                │   │   ├── package.yaml
+                │   │   └── src
+                │   │       └── <NAME>.hs
+                │   └── ...
+                ├── .docs
+                │   ├── instructions.md
+                │   ├── introduction.md
+                │   └── hints.md
+                └── .meta
+                    ├── config.json
+                    └── design.md
+```
+
+## Step 1: Add code files
+
+These are files specific to the Haskell track:
+
+- `package.yaml`: The Stack project file.
+- `stack.yaml`: The Stack configuration file.
+- `src/<NAME>.hs`: The stub file being handed to the student.
+- `test/Tests.hs`: The test suite being handed to the student.
+- `exemplars/success-...`: A directory containing an exemplar solution.
+  - `package.yaml`: That example solution's Stack project file.
+  - `src/<NAME>.hs>`: That example solution's source code.
+
+## Step 2: Add documentation files
+
+For the Markdown files in `.docs/` and `.meta/`, see the general advice on [how to implement a concept exercise][how-to-implement-a-concept-exercise].
+
+## Inspiration
+
+When implementing an exercise, it can be very useful to look at already implemented Haskell exercises. You can also check the exercise's [general concepts documents][reference] to see if other languages have already implemented an exercise for that concept.
+
+[reference]: https://github.com/exercism/v3/blob/main/reference
+[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/main/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
+[docs-concept-exercises]: https://github.com/exercism/v3/blob/main/docs/concept-exercises.md
+[docs-rationale-for-v3]: https://github.com/exercism/v3/blob/main/docs/rationale-for-v3.md
+[docs-features-of-v3]: https://github.com/exercism/v3/blob/main/docs/features-of-v3.md
+[anatomy-of-a-concept-exercise]: https://www.youtube.com/watch?v=gkbBqd7hPrA
+[spring-2013]: https://www.seas.upenn.edu/~cis194/spring13/
+[fall-2016]: https://www.seas.upenn.edu/~cis194/fall16/
```

Comment:
Some of these links are broken.

Reply:
We'll look into this!

Reply:
I've rephrased this file so that the broken links are not referenced.